strawberry/relay/mutations.py

```diff
@@ -0,0 +1,232 @@
+
+        return super().__call__(resolver)
+
+    @property
+    def arguments(self) -> List[StrawberryArgument]:
+        return list(self.default_args.values())
+
+    def get_result(
+        self,
+        source: Any,
+        info: Optional[Info],
+        args: List[Any],
+        kwargs: Dict[str, Any],
+    ) -> Union[Awaitable[Any], Any]:
+        assert self.base_resolver
+        input_obj = kwargs.pop("input")
+        return self.base_resolver(*args, **kwargs, **vars(input_obj))
+
+
+@overload
+def input_mutation(
+    *,
+    resolver: Callable[[], _T],
+    name: Optional[str] = None,
+    is_subscription: bool = False,
+    description: Optional[str] = None,
+    init: Literal[False] = False,
+    permission_classes: Optional[List[Type[BasePermission]]] = None,
+    deprecation_reason: Optional[str] = None,
+    default: Any = dataclasses.MISSING,
+    default_factory: Union[Callable[..., object], object] = dataclasses.MISSING,
+    metadata: Optional[Mapping[Any, Any]] = None,
+    directives: Optional[Sequence[object]] = (),
+) -> _T:
+    ...
+
+
+@overload
+def input_mutation(
+    *,
+    name: Optional[str] = None,
+    is_subscription: bool = False,
+    description: Optional[str] = None,
+    init: Literal[True] = True,
+    permission_classes: Optional[List[Type[BasePermission]]] = None,
+    deprecation_reason: Optional[str] = None,
+    default: Any = dataclasses.MISSING,
+    default_factory: Union[Callable[..., object], object] = dataclasses.MISSING,
+    metadata: Optional[Mapping[Any, Any]] = None,
+    directives: Optional[Sequence[object]] = (),
+) -> Any:
+    ...
+
+
+@overload
+def input_mutation(
+    resolver: Union[StrawberryResolver, Callable, staticmethod, classmethod],
+    *,
+    name: Optional[str] = None,
+    is_subscription: bool = False,
+    description: Optional[str] = None,
+    permission_classes: Optional[List[Type[BasePermission]]] = None,
+    deprecation_reason: Optional[str] = None,
+    default: Any = dataclasses.MISSING,
+    default_factory: Union[Callable[..., object], object] = dataclasses.MISSING,
+    metadata: Optional[Mapping[Any, Any]] = None,
+    directives: Optional[Sequence[object]] = (),
+) -> InputMutationField:
+    ...
+
+
+def input_mutation(
+    resolver=None,
+    *,
+    name: Optional[str] = None,
+    is_subscription: bool = False,
+    description: Optional[str] = None,
+    permission_classes: Optional[List[Type[BasePermission]]] = None,
+    deprecation_reason: Optional[str] = None,
+    default: Any = dataclasses.MISSING,
+    default_factory: Union[Callable[..., object], object] = dataclasses.MISSING,
+    metadata: Optional[Mapping[Any, Any]] = None,
+    directives: Optional[Sequence[object]] = (),
+    # This init parameter is used by pyright to determine whether this field
+    # is added in the constructor or not. It is not used to change
+    # any behavior at the moment.
+    init: Literal[True, False, None] = None,
+) -> Any:
```

Comment:
let's move this into another PR and decide if we should keep it or not 😊

Reply:
Sure! Will do that

Reply:
Done! Will open another PR soon

Should it go to `strawberry/mutation.py` as an alternative mutation approach?